chore: add development section (#261)
sourcegraph/sg.nvim: README.md

```lua
{
  --- The dir specified here is the absolute path to the sg.nvim repository
  dir = "~/code/path-to-sg-nvim-repo",
  dependencies = { "nvim-lua/plenary.nvim" }
}
```

### Dynamically switch to loading the plugin from the repository

For ease for development it can be useful to automatically switch to loading the plugin from this repository if we enter this directory. We can do this by doing the following:

1. In your plugin manager configuration, create the following function

```lua
local function load_sg()
  if vim.env.SG_NVIM_DEV then
    return { dir = vim.fn.getcwd(), dependencies = { "nvim-lua/plenary.nvim" } }
  else
    --- This is the configuration that lazy.nvim expects, but you can change it to whatever configuration your plugin manager expects
    return {
      "sourcegraph/sg.nvim",
      dependencies = { "nvim-lua/plenary.nvim" },
    }
  end
end
```
2. Update your configuration - example (lazy.nvim):

```lua
require("lazy").setup({
  "example/other-plugin",
  load_sg(),
}
```

> 3. Finally, we need the `SG_NVIM_DEV` variable to exist in our environment as soon as we enter the repository. We can do that by using [direnv](https://direnv.net/) which automatically loads `.envrc` is it exists. Let's edit the present `.envrc`

Author: ```suggestion
3. Finally, we need the `SG_NVIM_DEV` variable to exist in our environment as soon as we enter the repository. We can do that by using [direnv](https://direnv.net/) which automatically loads `.envrc` if it exists. Let's edit the current `.envrc`
```
